consensus/src/detail/mod.rs

```diff
@@ -202,6 +203,34 @@ impl Consensus {
         let block = consensus_ref.db_tx.get_block(id).map_err(BlockError::from)?;
         Ok(block)
     }
+
+    pub fn get_header_from_height(
+        &self,
+        height: &BlockHeight,
+    ) -> Result<Option<BlockHeader>, BlockError> {
+        let id = self.get_block_id_from_height(height)?.ok_or(BlockError::NotFound)?;
+        Ok(Some(
+            self.get_block(id)?.ok_or(BlockError::NotFound)?.header().clone(),
+        ))
+    }
+
+    pub fn get_locator(&self) -> Result<Vec<BlockHeader>, BlockError> {
+        let id = self.get_best_block_id()?.ok_or(BlockError::NotFound)?;
```

Comment:
Best block must be existed, otherwise we see a database corruption. It's legal to use here `expect("...")`

Reply:
Thanks!